test/DGmethods/advection_diffusion/advection_diffusion_model.jl

```diff
@@ -1,47 +1,80 @@
 using StaticArrays
 using CLIMA.VariableTemplates
-import CLIMA.DGmethods: BalanceLaw,
-                        vars_aux, vars_state, vars_gradient, vars_diffusive,
-                        flux_nondiffusive!, flux_diffusive!, source!,
-                        gradvariables!, diffusive!,
-                        init_aux!, init_state!,
-                        boundary_state!, wavespeed, LocalGeometry,
-                        num_state, num_gradient
-using CLIMA.DGmethods.NumericalFluxes: NumericalFluxNonDiffusive,
-                                       NumericalFluxDiffusive,
-                                       NumericalFluxGradient
+import CLIMA.DGmethods:
+    BalanceLaw,
+    vars_aux,
+    vars_state,
+    vars_gradient,
+    vars_diffusive,
+    flux_nondiffusive!,
+    flux_diffusive!,
+    source!,
+    gradvariables!,
+    diffusive!,
+    init_aux!,
+    update_aux!,
+    init_state!,
+    boundary_state!,
+    wavespeed,
+    LocalGeometry,
+    num_state,
+    num_gradient
+using CLIMA.DGmethods.NumericalFluxes:
+    NumericalFluxNonDiffusive, NumericalFluxDiffusive, NumericalFluxGradient
 import CLIMA.DGmethods.NumericalFluxes: boundary_flux_diffusive!
 
 abstract type AdvectionDiffusionProblem end
-struct AdvectionDiffusion{dim, P, fluxBC} <: BalanceLaw
-  problem::P
-  function AdvectionDiffusion{dim}(problem::P
-                                  ) where {dim, P <: AdvectionDiffusionProblem}
-    new{dim, P, false}(problem)
-  end
-  function AdvectionDiffusion{dim, fluxBC}(problem::P
-                                  ) where {dim, P <: AdvectionDiffusionProblem,
-                                           fluxBC}
-    new{dim, P, fluxBC}(problem)
-  end
+struct AdvectionDiffusion{dim, P, fluxBC, no_diffusion} <: BalanceLaw
```

Comment:
Only nitpick thing would be that instead of `no_diffusion` use `has_diffusion`, but really I'm fine either way.

Reply:
In this case would you prefer `has_diffusion` to be `false` by default  ? I also thought that this flag could be automatically set based on the user providing either `AdvectionProblem` or `AdvectionDiffusionProblem`, what do you think of this solution ?

Reply:
Leave it as is. I guess since its type is `AdvectionDiffusion` that diffusion off is the special case.